intake/catalog/base.py

```diff
@@ -364,6 +381,24 @@ def __init__(self, url, http_args={}, page_size=None, **kwargs):
     def _make_entries_container(self):
         return Entries(self)
 
+    def __dir__(self):
+        # Include (cached) tab-completable entries and normal attributes.
+        warnings.warn(
```

Comment:
I think this should only appear in the case that we know there are more entries than we are going to show, otherwise could be quite annoying

Reply:
Good point! Fixed and thoroughly tested.